src/plugins/vis_type_timelion/common/parser.ts

```diff
@@ -48,3 +48,12 @@ export interface ParsedExpression {
 }
 
 export const parseTimelionExpression = (input: string): ParsedExpression => parse(input);
+
+export const getIndexArgs = (expression: string) => {
```

Comment:
what if getIndexArgs will be renamed and return string array 

```
export const extractIndexesFromExpression = (expression: string) =>
  (parseTimelionExpression(expression)?.args ?? []).reduce<string[]>(
    (acc, { value, type, name, function: fn }) => {
      if (type === 'namedArg' && fn === 'es' && name === 'index' && value) {
        acc.push(value.text);
      }

      return acc;
    },
    []
  );
```

Reply:
done